File: leet code practice/Leaf-Similar Trees.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def leafSimilar(self, root1: TreeNode, root2: TreeNode) -> bool:
        arr1=[]
        arr2=[]
            
        self.leafvalues(root1,arr1)
        self.leafvalues(root2,arr2)
        
        if len(arr1)!=len(arr2):
            return False
        for i in range(len(arr1)):
            if arr1[i]!=arr2[i]:
                return False
        return True
        
    def leafvalues(self,root,arr):
        if not root:
             return 
        if not root.left and not root.right:
             return arr.append(root.val)
        else:
            self.leafvalues(root.left,arr)
            self.leafvalues(root.right,arr) 


            
        

File: leet code practice/Leaf-Similar Trees.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root1
 * @param {TreeNode} root2
 * @return {boolean}
 */

var leafSimilar = function(root1, root2) {
    
    var arr1=[]
    var arr2=[]
    
    leafvalues(root1,arr1)
    leafvalues(root2,arr2)
        
        console.log(arr1,arr2)
            if(arr1.length != arr2.length) return false;
        for(let i=0; i < arr1.length; ++i) {
            if(arr1[i] != arr2[i]) return false;
        }
        return true;
        
    };
        
        
    const leafvalues=(root,arr)=>{
        if(!root)
            return 
        if(root.left===null && root.right===null)
            return arr.push(root.val)
        else{
           return leafvalues(root.left,arr)+leafvalues(root.right,arr)
        }
        
    }
    
        
        